src/compiler/crystal/interpreter/compiler.cr

```diff
@@ -1099,7 +1099,7 @@ class Crystal::Repl::Compiler < Crystal::Visitor
   end
 
   private def dispatch_class_var(owner : Type, metaclass : Bool, node : ASTNode, &)
-    types = owner.all_subclasses.select { |t| t.is_a?(ClassVarContainer) }
+    types = owner.all_subclasses.select(ClassVarContainer)
```

Comment:
This is failing because `types` _must_ be `Array(Crystal::Type)` because we also add `owner` to it.
But `select(ClassVarContainer)` would return `Array(Crystal::ClassVarContainer)`.

It's probably best to refactor this a bit to make the typing more explicit.